addons/web/static/src/core/network/rpc_service.js

```diff
@@ -95,11 +95,35 @@ function jsonrpc(env, rpcId, url, params, settings = {}) {
 // -----------------------------------------------------------------------------
 // RPC service
 // -----------------------------------------------------------------------------
+String.prototype.hashCode = function() {
+    var hash = 0, i, chr;
+    if (this.length === 0) return hash;
+    for (i = 0; i < this.length; i++) {
+        chr   = this.charCodeAt(i);
+        hash  = ((hash << 5) - hash) + chr;
+        hash |= 0; // Convert to 32bit integer
+    }
+    return hash;
+};
 export const rpcService = {
     async: true,
     start(env) {
         let rpcId = 0;
         return function rpc(route, params = {}, settings) {
+            if (route == "/web/dataset/search_read" && window.localStorage['fast-menu-opening-proof-of-concept']) {
```

Comment:
are you sure about this? it looks very wrong to put records data in a cache. also, this PR looks non trivial, don't hesitate to come in rd-framework-js to explain what you are trying to do, so we can help you

Reply:
This is supposed to be short term cache (few seconds)